src/adapters/gmx/products/glp/gmxGlpAdapter.ts

```diff
@@ -373,4 +359,76 @@ export class GMXGlpAdapter implements IProtocolAdapter, IMetadataBuilder {
   async getRewardApy(_input: GetApyInput): Promise<ProtocolTokenApy> {
     throw new NotImplementedError()
   }
+
+  async getDeposits({
+    userAddress,
+    protocolTokenAddress,
+    fromBlock,
+    toBlock,
+  }: GetEventsInput): Promise<MovementsByBlock[]> {
+    return await this.getMovements({
+      protocolToken: await this.fetchProtocolTokenMetadata(
+        protocolTokenAddress,
+      ),
+      underlyingTokens: await this.fetchUnderlyingTokensMetadata(
+        protocolTokenAddress,
+      ),
+      filter: {
+        smartContractAddress: protocolTokenAddress,
+        fromBlock,
+        toBlock,
+        from: ZERO_ADDRESS,
+        to: userAddress,
+      },
+    })
+  }
+
+  async getWithdrawals({
+    userAddress,
+    protocolTokenAddress,
+    fromBlock,
+    toBlock,
+  }: GetEventsInput): Promise<MovementsByBlock[]> {
+    return await this.getMovements({
+      protocolToken: await this.fetchProtocolTokenMetadata(
+        protocolTokenAddress,
+      ),
+      underlyingTokens: await this.fetchUnderlyingTokensMetadata(
+        protocolTokenAddress,
+      ),
+      filter: {
+        smartContractAddress: protocolTokenAddress,
+        fromBlock,
+        toBlock,
+        from: userAddress,
+        to: ZERO_ADDRESS,
+      },
+    })
+  }
+
+  protected async fetchProtocolTokenMetadata(
+    _protocolTokenAddress: string,
+  ): Promise<Erc20Metadata> {
+    return (await this.buildMetadata()).protocolToken
+  }
+
+  protected async fetchUnderlyingTokensMetadata(
+    _protocolTokenAddress: string,
+  ): Promise<Erc20Metadata[]> {
+    return (await this.buildMetadata()).underlyingTokens
+  }
+
+  protected getUnderlyingTokenBalances(_input: {
+    userAddress: string
+    protocolTokenBalance: TokenBalance
+    blockNumber?: number | undefined
+  }): Promise<Underlying[]> {
+    throw new NotImplementedError()
+  }
+  protected getUnderlyingTokenConversionRate(
+    _protocolTokenMetadata: Erc20Metadata,
+    _blockNumber?: number | undefined,
+  ): Promise<UnderlyingTokenRate[]> {
+    throw new NotImplementedError()
+  }
```

Comment:
We are going to have a problem with `SimplePoolAdapter`.

In this case, I want to extend because I want to use its `getMovements`.
On the other hand, I don't want to use it for `getPositions` or `getPrices`, but the class forces me to implement this methods.

We might have to separate things a bit so that we can cherrypick. We can't have multiple inheritance, but we can have classes that partially implement some methods.

Reply:
ok a solution inst jumping out to me atm, perhaps we want to separate the profit methods e.g. getProfits and getMovements into a separate class.....not sure